lib/elixir/lib/module/types/infer.ex

```diff
@@ -402,31 +420,69 @@ defmodule Module.Types.Infer do
     false
   end
 
+  @doc """
+  Checks if the type has a type var.
+  """
+  def has_unbound_var?({:var, var}, context) do
+    case context.types do
+      %{^var => :unbound} -> true
+      %{^var => type} -> has_unbound_var?(type, context)
+    end
+  end
+
+  def has_unbound_var?({:tuple, args}, context),
+    do: Enum.any?(args, &has_unbound_var?(&1, context))
+
+  def has_unbound_var?({:union, args}, context),
+    do: Enum.any?(args, &has_unbound_var?(&1, context))
```

Comment:
How come we only check unions and types? What about the other complex types?

Reply:
I used subtype? as a reference so if something is missing here, it is missing there. What are they? lists and maps?

Reply:
We should sit together and document the types and which functions we have to implement for each new type. :)